src/index.js

```diff
@@ -157,6 +157,10 @@ class UglifyJsPlugin {
       uglifyOptions.output = uglifyOptions.output || {};
     }
 
+    if (typeof this.options.sourceMap === 'undefined' && ['sourcemap', 'source-map', 'hidden-source-map'].indexOf(compiler.options.devtool) > -1) {
```

Comment:
And if I missed one `if (typeof this.options.sourceMap === "undefined")` => `if (!this.options.sourceMap)` ?